arangod/Agency/Node.cpp

```diff
@@ -133,15 +138,22 @@ Node::Node(Node const& other)
     : _nodeName(other._nodeName),
       _parent(nullptr),
       _store(nullptr),
+      _children(nullptr),
       _ttl(other._ttl),
-      _value(other._value),
-      _vecBuf(other._vecBuf),
+      _value(nullptr),
+      _vecBuf(other._vecBuf ? std::make_unique<SmallBuffer>(*other._vecBuf) : nullptr),
       _vecBufDirty(other._vecBufDirty),
       _isArray(other._isArray) {
-  for (auto const& p : other._children) {
-    auto copy = std::make_shared<Node>(*p.second);
-    copy->_parent = this;  // new children have us as _parent!
-    _children.insert(std::make_pair(p.first, copy));
+  if (other._children) {
+    _children = std::make_unique<Children>();
```

Comment:
We could reserve enough space for all children to save a few vector reallocations:
```
_children->reserve(other._children->size());
```